### PR TITLE
Make Column Config work with Request V2

### PR DESCRIPTION
**Pull request by jkusa**
## Description

Make Column Config work with Request V2

## Proposed Changes

- Type action dispatcher and consumer
- Refactor fili grain formating
- Metric & Dimension selector for request v2
- Request converter to not add namespaces
- NaviColumnConfig to support request v2
- Move core types out of global.d.ts
- Reorganize types

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/navi-column-config.ts [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  isFiltered: boolean;
+  isRemovable: boolean;
+  fragment: ColumnFragment;
+};
+
+export default class NaviColumnConfig extends Component<NaviColumnConfigArgs> {
+  /**
+   * Dimension and metric columns from the request
+   */
+  @computed('args.report.request.{columns.@each.parameters,filters.[]}', 'args.report.visualization')
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/navi-column-config.ts:43]:
```suggestion
  @computed('args.report.request.{columns.@each.parameters,filters.[]}')
```

[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/navi-column-config.ts:43]: But thanks for asking, b/c that `visualization` ref is no longer needed :)

```suggestion
  @computed('args.report.request.{columns.@each.parameters,filters.[]}')
```

File: packages/reports/addon/components/navi-column-config.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      alias,
+      columnMetadata: { name },
+      hasParametersSet,
+      parameters
+    } = column;
+
+    if (alias) {
+      return alias;
+    }
+    return hasParametersSet ? `${name} (${Object.values(parameters).join(' ')})` : name;
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/navi-column-config.ts:99]: Could replace this with `naviFormatter.formatMetric(columnMetadata, parameters, alias)` (needs a rename)

File: packages/reports/addon/components/dimension-selector.js [hunk 3]
```diff
@@ -69,56 +63,12 @@ export default class DimensionSelector extends Component {
...
   get selectedColumns() {
-    if (!this.selectedTimeGrain) {
-      return this.selectedDimensions;
-    } else {
-      return arr([this.selectedTimeGrain, ...this.selectedDimensions]).uniq();
-    }
+    return this.request.dimensionColumns;
   }
 
   /*
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/dimension-selector.js:74]: Can we rename the `itemsChecked` prop?

File: packages/reports/addon/routes/reports/new.js [hunk 4]
```diff
@@ -96,27 +96,17 @@ export default Route.extend({
     let author = get(this, 'user').getUser();
 
     // Default to first data source + time grain
```
Comment (kevinhinterlong) [inline packages/reports/addon/routes/reports/new.js:98]: update comment

File: packages/reports/addon/components/dimension-selector.js (hunk 3 above)
Comment (jkusa) [reply packages/reports/addon/components/dimension-selector.js:74]: I can combine `selectedColumns` and `itemsChecked` and call it `selectedColumns`.

**Pull request merged**